onnxruntime/test/providers/cpu/tensor/resize_op_test.cc

```diff
@@ -2185,5 +2185,35 @@ TEST(ResizeOpTest, Antialias_Use_Extrapolation) {
       {4, 4, 4}, X, {3, 3, 3}, Y);
 }
 
+TEST(ResizeOpTest, TestOuputShapeRoundTripConversion) {
+  auto shape_check = [] (const std::vector<float>& X, const std::vector<float>& Y) {
+    int64_t N = X.size();
+    OpTester test("Resize", 13);
+    std::vector<float> scales{float(Y.size()) / float(X.size())};
+    std::vector<float> roi{};
+
+    test.AddAttribute("mode", "nearest");
+
+    test.AddInput<float>("X", {N}, X);
+    test.AddInput<float>("roi", {0}, roi);
+    test.AddInput<float>("scales", {1}, scales);
+    // Data of Y is random generated, because we don't care its correctness,
+    // just want to make sure its shape is correct.
+    test.AddOutput<float>("Y", {int64_t(Y.size())}, Y, false, 10000000.f, 100000000.f);
+    test.Run();
+  };
+
+  // Can we call UpsampleBase::computeOutputShape directly? It costs too much time.
+  constexpr int64_t in_size = 706;
+  for(int64_t out_size = 1; out_size < in_size; out_size += 3) {
+    std::vector<float> X(in_size);
+    std::vector<float> Y(out_size);
+    std::iota(X.begin(), X.end(), 1.f);
+    std::iota(Y.begin(), Y.end(), 1.f);
+    shape_check(X, Y);
+  }
```

Comment:
I would just put the python that was used to validate the asserting rounding is fine in a comment instead of doing all the iterations here as that will affect all test runs for minimal benefit.

and maybe the test could be simply taking input, converting to new size, taking the output from that and converting back to original size.